src/library/scala/concurrent/ExecutionContext.scala

```diff
@@ -166,14 +170,60 @@ object ExecutionContext {
     override final def reportFailure(t: Throwable): Unit = defaultReporter(t)
   }
 
+  /**
+   * This `ExecutionContext` uses the same thread pool as [[ExecutionContext.global]]. It attempts to batch nested
+   * task and execute them on the same thread as the enclosing task. This is ideally suited to execute short-lived tasks
+   * as it reduces the overhead of context switching.
+   *
+   * WARNING: long-running and/or blocking tasks should be demarcated within [[scala.concurrent.blocking]]-blocks
+   *          to ensure that any pending tasks in the current batch can be executed by another thread on `global`.
+   *
+   * === How to use ===
+   *
+   * This `ExecutionContext` is `private[scala]` to maintain binary compatibility. It was added in 2.13.3, code that
+   * uses it will fail to run with a 2.13.0-2 Scala library.
+   *
+   * Libraries should not reference this field because users of the library might be using an earlier Scala version.
```

Comment:
@lrytz Perhaps worth saying "Reference this field *directly*"?

Reply:
What is the additional information that you'd like to add? Is there an indirect way to access the field that is safe for a library?

Reply:
@lrytz I mean that it is safe to refer to the field indirectly (via reflection, var-handles etc) since that won't break bincompat (assuming that any error is handled at runtime, with a fallback)